algorithms/loopclosure/loop-closure-handler/src/loop-closure-handler.cc

```diff
@@ -240,10 +241,6 @@ bool LoopClosureHandler::handleLoopClosure(
       static_cast<unsigned int>(query_vertex_n_frame.getNumFrames()),
       query_vertex_landmark_ids.size());
 
-  // Make sure only one of those options is selected. We can't merge landmarks
```

Comment:
it is related to the lc / relax part i. the description, before the lc handler couldn't do both at the same time, now it can, so this xor check is not needed anymore. Basically before it was not possible because once you merge the landmarks the lc edges cannot be added since finding the best vertex to connect it to doesn't work, because the merged landmarks already created co-observability. So I rearranged the two and simplified the logic a bit.